core/kazoo_attachments/src/kz_att_s3.erl

```diff
@@ -78,11 +80,24 @@ get_file_path('undefined', DbName, DocId, AName) ->
 get_file_path(Path, DbName, DocId, AName) ->
     kz_util:to_list(list_to_binary([Path, "/", DbName, "/", DocId, "_", AName])).
 
--spec get_s3_values(ne_binary()) -> {ne_binary(), ne_binary(), ne_binary(), ne_binary(), api_binary()}.
-get_s3_values(S3) ->
+-spec get_s3_values(ne_binary(), kz_data:connection()) ->
+                           {ne_binary(), ne_binary(), ne_binary(), ne_binary(), api_binary()}.
+get_s3_values(S3, 'undefined') ->
     case binary_to_term(base64:decode(S3)) of
         {Key, Secret, Bucket, Path} ->
             {Key, Secret, ?AMAZON_S3_HOST, Bucket, Path};
         {_Key, _Secret, _Host, _Bucket, _Path} = V2 ->
             V2
+    end;
+get_s3_values(S3, HandlerProps) ->
+    {_Key, _Secret, Host, Bucket, Path} = get_s3_values(S3, 'undefined'),
+    try get_map_values(HandlerProps) of
+        {Bucket, Key, Secret, _Path, Host} ->
+            {Key, Secret, Host, Bucket, Path};
+        {_Bucket, Key, Secret, _Path, _Host} ->
+            {_Key, _Secret, Host, Bucket, Path}
+    catch
+        _E:_R ->
+            lager:info("unable to get s3 cred value using att values (~s): ~p", [_E, _R]),
+            {_Key, _Secret, Host, Bucket, Path}
```

Comment:
@tsromana we usualy use `underscore` for variables that we really don't use besides logging
how about
```
    Original =  {_, _, Host, Bucket, Path} = get_s3_values(S3, 'undefined'),
    try get_map_values(HandlerProps) of
        {Bucket, Key, Secret, _Path, Host} ->
            {Key, Secret, Host, Bucket, Path};
        _ -> Original
    catch
...
```